internal/kubernetes/operator/deployment/deployment_test.go

```diff
@@ -38,6 +38,11 @@ type MockAtlasOperatorClusterStore struct {
 	projectToClusterToSchedule            map[string]map[string]*mongodbatlas.CloudProviderSnapshotBackupPolicy
 	projectToServerlessClusters           map[string]map[string]*mongodbatlas.Cluster
 	projectIDToServerlessPrivateEndpoints map[string]map[string][]mongodbatlas.ServerlessPrivateEndpointConnection
+	projectIDToGlobalCluster              map[string]map[string]*mongodbatlas.GlobalCluster
+}
+
+func (m *MockAtlasOperatorClusterStore) GlobalDeployment(projectID string, instanceName string) (*mongodbatlas.GlobalCluster, error) {
+	return m.projectIDToGlobalCluster[projectID][instanceName], nil
```

Comment:
no need to define this mock. We use `mockgen` and you already added the correct mock annotation in internal/store/global_deployment.go. So if you run `make gen-mocks`, the mock is autogenerated and can be used in your test

Reply:
Let's merge it as is. But we will create a ticket to rewrite our tests with autogenerated mocks. Is it ok?

Reply:
sure, can you share a link to the ticket and ETA? just want to make sure that this will be addressed

Reply:
https://jira.mongodb.org/browse/CLOUDP-152961 
I will do this task as soon as I finish the current one. So I will create a PR with a test update on Friday or Monday.

Reply:
Thanks!